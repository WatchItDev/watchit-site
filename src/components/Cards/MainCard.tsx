// REACT IMPORTS
import React, { ReactElement } from 'react'

// MUI IMPORTS
import { Card, CardContent, CardHeader, CardProps, Divider, Theme, useTheme } from '@mui/material'
import { SxProps } from '@mui/system'

/* eslint-disable  @typescript-eslint/strict-boolean-expressions */

// ===========================|| CARD - MAIN ||=========================== //

export interface MainCardProps extends Omit<CardProps, 'title'> {
  border?: boolean
  boxShadow?: boolean
  content?: boolean
  contentSx?: SxProps<Theme>
  secondary?: React.ReactNode | string
  shadow?: string
  sx?: SxProps<Theme>
  title?: ReactElement | string
}

const MainCard = React.forwardRef<HTMLDivElement, MainCardProps>(({ border, content, contentSx, boxShadow, title, ...props }, ref) => {
  const theme = useTheme()

  return (
    <Card
      {...props}
      ref={ref}
      sx={{
        border: border ? '1px solid' : 'none',
        borderColor: theme.palette.primary.light,
        ':hover': {
          boxShadow: boxShadow ? props.shadow ?? '0 2px 14px 0 rgb(32 40 45 / 8%)' : 'inherit'
        },
        ...props.sx
      }}
    >
      {/* card header and action */}
      {title && <CardHeader sx={{ p: 2, '& .MuiCardHeader-action': { mr: 0 } }} title={title} action={props.secondary} />}

      {/* content & header divider */}
      {title && <Divider />}

      {/* card content */}
      {content && (
        <CardContent sx={contentSx}>
          {props.children}
        </CardContent>
      )}
      {!content && props.children}
    </Card>
  )
})

export default MainCard
