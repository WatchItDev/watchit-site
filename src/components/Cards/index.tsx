// ===========================|| CARD EXPORTS ||=========================== //
export { default as MainCard } from '@components/Cards/MainCard'
