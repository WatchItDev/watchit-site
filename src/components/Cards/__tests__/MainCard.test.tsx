import { MainCard } from '@components/Cards'
import { shallow } from 'enzyme'
import React from 'react'

describe('<MainCard />', () => {
  it('should render', () => {
    const component = shallow(<MainCard />)
    expect(component).toMatchSnapshot()
  })
})
