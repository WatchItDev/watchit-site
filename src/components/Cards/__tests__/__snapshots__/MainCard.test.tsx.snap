// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainCard /> should render 1`] = `ShallowWrapper {}`;
